bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php

```diff
@@ -175,6 +177,14 @@ public function __wakeup()
 
         // set current cart
         foreach ($this->getItems() as $item) {
+
+            // if a product is deleted in backend, then remove item from SessionCart
+            $product = DataObject::getById($item->getProductId());
```

Comment:
we should call `getProduct()` here ... then also PHPStan is happy ...

Reply:
done.

Reply:
Changed approach here as it impacts the performance coz when updating cart, it checks for product exists by calling `DataObject::getById()` which loads the product. 

so now we use `MockProduct` for CartItem in getProduct() calls when actual product doesn't exist.